lib/Model/Message.php

```diff
@@ -222,6 +223,25 @@ public function getLocalAttachments(): array {
 		return $this->localAttachments;
 	}
 
+	/**
+	 * @param Attachment $attachment
+	 *
+	 * @return void
+	 *
+	 * Adds a file that's coming from another email's attachment (typical
+	 * use case is forwarding a message)
+	 *
+	 */
+	public function addAttachmentFromAttachment(Attachment $attachment) {
+		$part = new Horde_Mime_Part();
+		$part->setCharset('us-ascii');
+		$part->setDisposition('attachment');
+		$part->setName($attachment->getName());
+		$part->setContents($attachment->getContents());
+		$part->setType($attachment->getType());
+		$this->cloudAttachments[] = $part;
```

Comment:
:thinking: my IDE says `$this->cloudAttachments` is only read via `\OCA\Mail\Model\Message::getCloudAttachments` and that method is only used in tests.

Reply:
I don't understand your comment. But, I must say I don't really understand this cloudAttachments vs localAttachments distinction: It doesn't seem to make much sense

Reply:
https://github.com/nextcloud/mail/pull/3950#pullrequestreview-553717905 I was wrong :wink: